desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,672 @@
+    the complex plane of Fourier series make it preferable in practice to choose
+    coordinate systems such that the function to approximate is periodic. The
+    Chebyshev series is preferred to other orthogonal polynomial series since
+    fast discrete polynomial transforms (DPT) are implemented via fast transform
+    to Chebyshev then DCT. Although nothing prohibits a direct DPT, we want to
+    rely on existing, optimized libraries. There are other reasons to prefer
+    Chebyshev series not discussed here. Therefore, |B| is interpolated to a
+    Fourier-Chebyshev series in (α, ζ).
+
+    Computing accurate series expansions in (α, ζ) coordinates demands
+    particular interpolation points in that coordinate system. Newton iteration
+    is used to compute θ at these interpolation points. Note that interpolation
+    is necessary because there is no transformation that converts series
+    coefficients in periodic coordinates, e.g. (ϑ, ϕ), to a low order
+    polynomial basis in non-periodic coordinates. For example, one can obtain
+    series coefficients in (α, ϕ) coordinates from those in (ϑ, ϕ) as follows
+        g : ϑ, ϕ ↦ ∑ₘₙ aₘₙ exp(j [mϑ + nϕ])
+
+        g : α, ϕ ↦ ∑ₘₙ aₘₙ exp(j [mα + (m ι + n)ϕ])
+    However, the basis for the latter are trigonometric functions with
+    irrational frequencies, courtesy of the irrational rotational transform.
+    Globally convergent root-finding schemes for that basis (at fixed α) are
+    not known. The denominator of a close rational could be absorbed into the
+    coordinate ϕ, but this balloons the frequency, and hence the degree of the
+    series.
+
+    One may note that because Fourier series converge faster than Chebyshev,
+    an alternate strategy is to interpolate |B| to a double Fourier series in
+    (ϑ, ϕ), then apply bisection methods to find roots of f with mesh size
+    inversely proportional to the max frequency along the field line: M ι + N.
+    ``Bounce2D`` does not use that approach because that root-finding scheme is
+    inferior. Note that if the Chebyshev series convergence is not satisfactory,
+    we can implement a change of variables for the Chebyshev interpolation that
+    may allow earlier truncation of the series without loss of accuracy. Another
+    option with faster convergence still is to use a filtered Fourier series
+    with a non-periodic basis in the Gibbs layer. An example is given at
+    doi.org/10.1016/j.aml.2006.10.001. Empirically, we observe that the
+    Chebyshev series has satisfactory convergence. It is particularly fast for
+    θ(α, ζ). Furthermore, the Fourier series converges rapidly for |B|(α, ζ)
+    in near omnigenous configurations because (∂|B|/∂α)|ρ,ζ vanishes. However,
+    the Fourier series of θ(α, ζ) converges slower than desired; we must accept
+    this because no basis supports a larger convergence region in the complex
+    plane than do Fourier series.
```

Comment:
Idea:
For unoptimized |B| is typically higher `N` to capture all oscillations (at least if `Bounce1D`). θ will likely always be fine with low `N` since it's essentially just proportional to iota at fixed alpha, so Chebyshev converges fast.
Rather than doing coordinate mapping on `M * N` tensor grid each large, do it on `M * (N_{\theta}` where $N_{\theta}$ is small. Then use the resulting, accurate, θ(α, ζ) to compute the interpolation points in (α, ζ) so that |B| (α, ζ) can be constructed with large Chebyshev resolution. This reduces the number of points to perform root finding on each time we want to compute a neoclassical objective.